test/unit/resources/mock/iam/aws_iam_saml_provider_mock.rb

```diff
@@ -0,0 +1,25 @@
+require_relative '../aws_base_resource_mock'
+
+class AwsIamSamlProviderMock < AwsBaseResourceMock
+
+  # Default attributes.
+  def initialize
+    super
+    @provider = OpenStruct.new(
```

Comment:
I like this as long as it's what we expect from the API.

Reply:
Yup, API provides a struct